File: boj/problem2226/Main.java
package problem2226;

import java.io.BufferedReader;
import java.io.BufferedWriter;
import java.io.IOException;
import java.io.InputStreamReader;
import java.io.OutputStreamWriter;
import java.math.BigInteger;
import java.util.StringTokenizer;

public class Main {
    public static String solve(int N) {
        BigInteger answer = BigInteger.valueOf(0);
        BigInteger[] oneStartArr = {BigInteger.valueOf(0), BigInteger.valueOf(1)};
        BigInteger[] zeroStartArr = {BigInteger.valueOf(0), BigInteger.valueOf(1)};

        if (N == 1 || N == 2)
            return oneStartArr[N - 1].toString();

        for (int i = 2; i < N; i++) {
            BigInteger prevOneStartZeroCnt = oneStartArr[1];
            BigInteger prevZeroStartZeroCnt = zeroStartArr[1];
            if ((i - 1) % 2 != 0) {
                oneStartArr[1] = prevOneStartZeroCnt.add(prevZeroStartZeroCnt)
                        .subtract(BigInteger.valueOf(1));
                oneStartArr[0] = prevOneStartZeroCnt;
                zeroStartArr[1] = prevOneStartZeroCnt.add(prevZeroStartZeroCnt);
                zeroStartArr[0] = prevZeroStartZeroCnt;
            } else {
                oneStartArr[1] = prevOneStartZeroCnt.add(prevZeroStartZeroCnt);
                oneStartArr[0] = prevOneStartZeroCnt;
                zeroStartArr[1] = prevOneStartZeroCnt.add(prevZeroStartZeroCnt);
                zeroStartArr[0] = prevZeroStartZeroCnt;
            }
        }

        answer = oneStartArr[1];

        return answer.toString();
    }

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());

        int N = Integer.parseInt(st.nextToken());

        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
        bw.write(solve(N));
        bw.flush();
    }
}

File: boj/problem2226/main.py
def solve(N: int) -> str:
    answer = 0
    one_start_arr = [0, 1]
    zero_start_arr = [0, 1]

    if N == 1 or N == 2:
        return str(one_start_arr[N - 1])

    for i in range(2, N, 1):
        prev_one_start_zero_cnt = one_start_arr[1]
        prev_zero_start_zero_cnt = zero_start_arr[1]

        if (i - 1) % 2 != 0:
            one_start_arr[1] = prev_one_start_zero_cnt + prev_zero_start_zero_cnt - 1
            one_start_arr[0] = prev_one_start_zero_cnt
            zero_start_arr[1] = prev_one_start_zero_cnt + prev_zero_start_zero_cnt
            zero_start_arr[0] = prev_zero_start_zero_cnt
        else:
            one_start_arr[1] = prev_one_start_zero_cnt + prev_zero_start_zero_cnt
            one_start_arr[0] = prev_one_start_zero_cnt
            zero_start_arr[1] = prev_one_start_zero_cnt + prev_zero_start_zero_cnt
            zero_start_arr[0] = prev_zero_start_zero_cnt

    answer = one_start_arr[1]

    return str(answer)


def main():
    N = int(input())

    print(solve(N))


if __name__ == "__main__":
    main()
